### PR TITLE
DOC: Cheatsheet change structure

### PR DESCRIPTION
**Pull request by eloycoto**
Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

### CODE REVIEW

**Comment by manalibhutiyani** [comment]
I had made a list as part of my docs list

```
Cilium monitor commands missing:
cilium monitor -t L7 missing
Many ct/bpf commands are missing.
Microscope commands?
cilium set config example

Cilium related K8s commands:
1. kubectl get cnp
2. Kubectl get cep.
More documentation/examples on `kubectl get cep`? 
Will appreciate a brief summary of what this command does?
3. Any more commands I am unaware of.
```

**Comment by eloycoto** [comment]
@manalibhutiyani ready to review it.

File: Documentation/cheatsheet.rst [hunk 1]
```diff
@@ -2,157 +2,396 @@
...
 
-	cilium policy get
 
+Get list of all imported policy rules
 ::
 
-	# Remove all policy
+	cilium policy get
+
+Remove all policy
```
Comment (nebril) [inline Documentation/cheatsheet.rst]: s/policy/policies

File: Documentation/cheatsheet.rst [hunk 2]
```diff
@@ -2,157 +2,396 @@
...
 
+Verbose output (including debug if enabled)
 ::
 
-        # Filter for only the events to endpoint
+    cilium monitor -v
 
-        cilium monitor --related-to=<id>
 
+Filter for only the events to endpoint
```
Comment (nebril) [inline Documentation/cheatsheet.rst]: s/events to endpoint/events related to endpoint

I would say that `events to endpoint` are handled by `--to` flag.

File: Documentation/cheatsheet.rst [hunk 3]
```diff
@@ -2,157 +2,396 @@
...
+Check cluster Connectivity
 ::
 
-        # Show recent endpoint specific log entries
+	cilium-health status
 
-        cilium endpoint log <id>
+There is also a `blog post
+<https://cilium.io/blog/2018/2/6/cilium-troubleshooting-cluster-health-monitor>`_
+related with this tool.
```
Comment (nebril) [inline Documentation/cheatsheet.rst]: s/related with/related to

File: Documentation/cheatsheet.rst [hunk 4]
```diff
@@ -2,157 +2,396 @@
...
 
-        # List node tunneling mapping information
+    sudo cilium bpf proxy list
+
+
+Kubernetes examples:
+=====================
+
+If you running Cilium on top of Kubernetes you may also want a way to list all
+cilium endpoints or policies from a single Kubectl commands. Cilium by using
```
Comment (nebril) [inline Documentation/cheatsheet.rst]: I would rephrase the sentence starting with `Cilium by using` to `Cilium provides all this information to the user by using (...)`

File: Documentation/cheatsheet.rst [hunk 5]
```diff
@@ -2,157 +2,396 @@
...
+`Kubernetes Resource Definitions
+<https://kubernetes.io/docs/concepts/api-extension/custom-resources/>`_ provides
+all this information to the user:
+
+
+Policies
+---------
+
+In Kubernetes you can use two kind of policies, Kubernetes Network Policies or
```
Comment (nebril) [inline Documentation/cheatsheet.rst]: s/kind/kinds

File: Documentation/cheatsheet.rst [hunk 6]
```diff
@@ -2,157 +2,396 @@
...
+cilium network polcies. Both can be retrieved from the ``kubectl`` command:
```
Comment (nebril) [inline Documentation/cheatsheet.rst]: s/cilium network polcies/Cilium Network Policies

File: Documentation/cheatsheet.rst [hunk 7]
```diff
@@ -2,157 +2,396 @@
...
+   :name: Kubernetes Cilium Policies
+   :caption: Kubernetes Cilium Policies
+
+    $ kubectl get cnp
+    NAME      AGE
+    rule1     3m
+    $ kubectl get cnp rule1
+    NAME      AGE
+    rule1     3m
+    $ kubectl get cnp rule1 -o json
```
Comment (nebril) [inline Documentation/cheatsheet.rst:343]: Should output be shown here?

File: Documentation/cheatsheet.rst [hunk 8]
```diff
@@ -2,157 +2,396 @@
...
+
+
+Endpoints
+----------
+
+To retrieve a list of all endpoints managed by cilium, a ``Cilum Endpoint``
+command can be used.
```
Comment (nebril) [inline Documentation/cheatsheet.rst]: s/command/resource

File: Documentation/cheatsheet.rst [hunk 9]
```diff
@@ -2,157 +2,396 @@
...
+    $ kubectl get cnp rule1
+    NAME      AGE
+    rule1     3m
+    $ kubectl get cnp rule1 -o json
+
+
+Endpoints
+----------
+
+To retrieve a list of all endpoints managed by cilium, a ``Cilum Endpoint``
```
Comment (nebril) [inline Documentation/cheatsheet.rst]: I would drop second `a` here.

File: Documentation/cheatsheet.rst [hunk 10]
```diff
@@ -2,157 +2,396 @@
...
+    $ kubectl get cep 700e0976-6cb50b02 -o json
+
+    $ kubectl get cep -o jsonpath='{range .items[*]}{@.status.id}{"="}{@.status.status.policy.spec.policy-enabled}{"\n"}{end}'
+    30391=ingress
+    5766=ingress
+    51796=none
+    40355=none
+
+
+Microscope
```
Comment (nebril) [inline Documentation/cheatsheet.rst:378]: Thanks for this short writeup.

**Review by nebril — CHANGES_REQUESTED** [review]
Some typos and nitpicks in review.

File: Documentation/cheatsheet.rst [hunk 11]
```diff
@@ -2,157 +2,396 @@
 Command Cheatsheet
 ******************
 
-Below is a short list of examples of the various commands Cilium has to offer.
-If a command you use is missing please don’t hesitate to add it to one of the
-groups or create a new one.
+Cilium is controlled via a easy command-line interface. This CLI is a single
```
Comment (raybejjani) [inline Documentation/cheatsheet.rst]: "an easy"

File: Documentation/cheatsheet.rst [hunk 12]
```diff
@@ -2,157 +2,396 @@
...
+retrieved from Cilium Daemon. If you need something more detailed you can use JSON
+output, to get the JSON output you can use the global option ``-o json``
 
 ::
 
-        # Check the status of the agent
+    $ cilium endpoint list -o json
 
-        cilium status
+Moreover, Cilium also provides a `JSONPAth
```
Comment (raybejjani) [inline Documentation/cheatsheet.rst]: "provides `JSONPath` support"

(no "a" and JSONPAth has an uppercase A)

File: Documentation/cheatsheet.rst [hunk 13]
```diff
@@ -2,157 +2,396 @@
...
+::
 
-	cilium policy import <my-policy.json>
+    cilium config
+
+Policy management
+-----------------
 
+
+Importing a policy
```
Comment (raybejjani) [inline Documentation/cheatsheet.rst]: Since this is a cheat sheet, it might be good to mention what kind of policy this imports.
"Import a Cilium Network Policy"
or a note saying it isn't a k8s NP

File: Documentation/cheatsheet.rst [hunk 14]
```diff
@@ -2,157 +2,396 @@
...
+Get list of all imported policy rules
 ::
 
-	# Remove all policy
+	cilium policy get
+
+Remove all policy
+::
 
 	cilium policy delete --all
```
Comment (raybejjani) [inline Documentation/cheatsheet.rst]: We can also delete policies by label. It might be worth having an example of that too.

File: Documentation/cheatsheet.rst [hunk 15]
```diff
@@ -2,157 +2,396 @@
...
+Endpoints
+----------
+
+To retrieve a list of all endpoints managed by cilium, a ``Cilum Endpoint``
+command can be used.
 
 ::
 
-        # Checking logs for verifier issue
+    $ kubectl get cep
```
Comment (raybejjani) [inline Documentation/cheatsheet.rst:354]: maybe add `--all-namespaces` as a subtle hint that these CRDs are namespace scoped.

File: Documentation/cheatsheet.rst [hunk 16]
```diff
@@ -2,157 +2,396 @@
...
+    5c6211b5-9217a4d1   1m
+    5c6211b5-dccc3d24   1m
+    700e0976-6cb50b02   3m
+    700e0976-afd3a30c   3m
+    78092a35-4874ed16   1m
+    78092a35-4b08b92b   1m
+    9b74f61f-14571299   7s
+    9b74f61f-f9a96f4a   7s
+
+    $ kubectl get cep 700e0976-6cb50b02 -o json
```
Comment (raybejjani) [inline Documentation/cheatsheet.rst:369]: Maybe break this out into another code section with a comment like:
"To view the raw JSON for processing, or filter via JSONPath:"

File: Documentation/cheatsheet.rst [hunk 17]
```diff
@@ -2,157 +2,396 @@
...
+    51796=none
+    40355=none
+
+
+Microscope
+----------
+
+Cilium also provides an option to monitor all connections from all Kubernetes
+nodes. `Microscope <https://github.com/cilium/microscope>`_ is a distributed monitor that connects to all Cilium instances
+and retrieve monitor information from there.
```
Comment (raybejjani) [inline Documentation/cheatsheet.rst]: retrieve -> retrieves

Comment (raybejjani) [reply Documentation/cheatsheet.rst]: I would rephrase this:
Cilium also provides the ability to monitor all cilium-managed connections in the kubernetes cluster via `Microscope <https://github.com/cilium/microscope>`_. It is a distributed monitor that connects to all Cilium instances and retrieves monitor information from each node.

File: Documentation/cheatsheet.rst [hunk 18]
```diff
@@ -2,157 +2,396 @@
...
+
+Microscope can be installed as a pod and runs from there, the basic usage is the
+following:
```
Comment (raybejjani) [inline Documentation/cheatsheet.rst]: "Microscope can be installed and run as a pod with:"

**Review by raybejjani — APPROVED** [review]
nits from me

File: Documentation/cheatsheet.rst (hunk 7 above)
Comment (eloycoto) [reply Documentation/cheatsheet.rst:343]: It is a bit verbose. I would only want to emphasize that the json ouput it's in there.

File: Documentation/cheatsheet.rst (hunk 15 above)
Comment (eloycoto) [reply Documentation/cheatsheet.rst:354]: I do not think so, it's a kubernetes command, you expect that no?

File: Documentation/cheatsheet.rst (hunk 16 above)
Comment (eloycoto) [reply Documentation/cheatsheet.rst:369]: I do not think so either. It's a kubectl command too, the jsonpath it's expected, only show a small example that it's supported, but a new section I think that it's not needed.

File: Documentation/cheatsheet.rst (hunk 15 above)
Comment (raybejjani) [reply Documentation/cheatsheet.rst:354]: for some reason I thought the CNPs were namespace agnostic. They are, in fact, namespaced so ignore me.

File: Documentation/cheatsheet.rst (hunk 16 above)
Comment (raybejjani) [reply Documentation/cheatsheet.rst:369]: ok

**Comment by eloycoto** [comment]
test-me-please

**Review by nebril — APPROVED** [review]
